### PR TITLE
WIP: Control Project Prometheus RBAC via injected SA

### PR DESCRIPTION
**Pull request by thxCode**
- Upgrade Monitoring chart to 0.0.3
- Project Monitoring will cause permission leak, we should use an injected SA to bootstrap all workload in project level.
- Add permission to kube-state exporter, fix rancher/rancher#18830, rancher/rancher#19218
- Fix https://github.com/rancher/rancher/issues/19381

### CODE REVIEW

**Comment by loganhz** [comment]
Link to https://github.com/rancher/rancher/issues/19381

File: charts/rancher-monitoring/v0.0.3/charts/operator/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+description: Creates Prometheus Operator for Kubernetes which maintaining by Rancher 2.
```
Comment (cjellick) [inline charts/rancher-monitoring/v0.0.3/charts/operator/Chart.yaml]: English is incorrect here

File: charts/rancher-monitoring/v0.0.3/charts/prometheus/templates/rbac.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.enabledRBAC (not .Values.serviceAccountName) }}
+{{- if and .Values.enabledRBAC (not .Values.global.rancherInjectedServiceAccountName) }}
```
Comment (cjellick) [inline charts/rancher-monitoring/v0.0.3/charts/prometheus/templates/rbac.yaml:1]: above, you added to this if check like:
```
{{- if and .Values.enabledRBAC (not .Values.serviceAccountName) (not .Values.global.rancherInjectedServiceAccountName) }}
```
how do you decide when to add an additional check vs replace the check

File: charts/rancher-monitoring/v0.0.3/charts/grafana/dashboards/w_prometheus_stats-dashboard.json [hunk 3]
```diff
@@ -0,0 +1 @@
...:{"list":[{"allValue":null,"current":{"text":"$job","value":"$job"},"datasource":"RANCHER_MONITORING","definition":"query_result(prometheus_tsdb_head_samples_appended_total)","hide":2,"includeAll":false,"label":null,"multi":false,"name":"job","options":[],"query":"query_result(prometheus_tsdb_head_samples_appended_total)","refresh":2,"regex":"/.*job=\"([^\"]+)/","skipUrlSync":false,"sort":1,"tagValuesQuery":"","tags":[],"tagsQuery":"","type":"query","useTags":false},{"allValue":null,"current":{"text":"cattle-prometheus","value":"cattle-prometheus"},"datasource":"RANCHER_MONITORING","definition":"query_result(up{job=~\"$job\"})","hide":0,"includeAll":false,"label":"Namespace","multi":false,"name":"namespace","options":[],"query":"query_result(up{job=~\"$job\"})","refresh":2,"regex":"/.*namespace=\"([^\"]+).*/","skipUrlSync":false,"sort":0,"tagValuesQuery":"","tags":[],"tagsQuery":"","type":"query","useTags":false}]},"time":{"from":"now-5m","to":"now"},"timepicker":{"refresh_intervals":["5s","10s","30s","1m","5m","15m","30m","1h","2h","1d"],"time_options":["5m","15m","1h","6h","12h","24h","2d","7d","30d"]},"timezone":"browser","title":"Prometheus Stats","uid":"4hEg-4eiz","version":1}
```
Comment (cjellick) [inline charts/rancher-monitoring/v0.0.3/charts/grafana/dashboards/w_prometheus_stats-dashboard.json]: can you explain what these giant json blobs are?

Comment (loganhz) [reply charts/rancher-monitoring/v0.0.3/charts/grafana/dashboards/w_prometheus_stats-dashboard.json]: Please remove the change for grafana dashboard. It's nice to have. But we should not do this for now. If you want to debug for users in 2.2.2, you can just ask users to import the grafana dashboard then.

**Comment by cjellick** [comment]
associated issues closed

**Pull request closed without merging**